lib/puppet/type/cisco_interface_ospf.rb

```diff
@@ -240,34 +246,33 @@ def name
   # and message_digest_password combination
 
   validate do
-    fail("area must be supplied when ensure=present") if
-      self[:ensure] == :present and self[:area].nil?
+    fail('area must be supplied when ensure=present') if
+      self[:ensure] == :present && self[:area].nil?
 
-    if (self[:message_digest_key_id].nil?) and
-         (!self[:message_digest_algorithm_type].nil? or
-          !self[:message_digest_encryption_type].nil? or
-          !self[:message_digest_password].nil?)
+    if (self[:message_digest_key_id].nil?) &&
+       (!self[:message_digest_algorithm_type].nil? ||
+        !self[:message_digest_encryption_type].nil? ||
+        !self[:message_digest_password].nil?)
       fail("If message_digest_key_id is not present in the manifest, \
             the following attributes must not be present - \
             message_digest_algorithm_type, \
             message_digest_encryption_type, message_digest_password" )
     end
 
-    if (!self[:message_digest_key_id].nil?) and
-         (self[:message_digest_algorithm_type].nil? or
-          self[:message_digest_encryption_type].nil? or
-          self[:message_digest_password].nil?)
+    if (!self[:message_digest_key_id].nil?) &&
+       (self[:message_digest_algorithm_type].nil? ||
+        self[:message_digest_encryption_type].nil? ||
+        self[:message_digest_password].nil?)
       fail("If message_digest_key_id is present in the manifest, \
-	    the following attributes must also be present - \
+      the following attributes must also be present - \
             message_digest_algorithm_type, \
             message_digest_encryption_type, message_digest_password")
     end
-    
-    if self[:passive_interface] and
-      !/^lo\S+$/.match(self[:interface].downcase).nil?
-      fail "passive_interface value cannot be set on loopback interfaces"
-    end
 
+    if self[:passive_interface] &&
+       !/^lo\S+$/.match(self[:interface].downcase).nil?
```

Comment:
Not specific to your change but `!regex.match(str).nil?` is equivalent to simply `regex.match(str)`